demos/mask_rcnn_demo/cpp/main.cpp

```diff
@@ -45,240 +49,327 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    logBasicModelInfo(model);
 
-        /** network dimensions for image input **/
-        const InferenceEngine::TensorDesc& inputDesc = inputInfo[imageInputName]->getTensorDesc();
-        IE_ASSERT(inputDesc.getDims().size() == 4);
-        size_t netBatchSize = getTensorBatch(inputDesc);
-        size_t netInputHeight = getTensorHeight(inputDesc);
-        size_t netInputWidth = getTensorWidth(inputDesc);
-
-        /** Collect images **/
-        std::vector<cv::Mat> images;
-
-        if (netBatchSize > imagePaths.size()) {
-            slog::warn << "Network batch size is greater than number of images (" << imagePaths.size() <<
-                       "), some input files will be duplicated" << slog::endl;
-        } else if (netBatchSize < imagePaths.size()) {
-            slog::warn << "Network batch size is less than number of images (" << imagePaths.size() <<
-                       "), some input files will be ignored" << slog::endl;
-        }
+    // Taking information about model inputs
+    ov::OutputVector inputs = model->inputs();
+    ov::OutputVector outputs = model->outputs();
 
-        auto startTime = std::chrono::steady_clock::now();
-        for (size_t i = 0, inputIndex = 0; i < netBatchSize; i++, inputIndex++) {
-            if (inputIndex >= imagePaths.size()) {
-                inputIndex = 0;
-            }
+    // Validate inputs/outputs
+    validateInputsAndOutputs(inputs, outputs);
 
-            cv::Mat image = cv::imread(imagePaths[inputIndex], cv::IMREAD_COLOR);
+    size_t modelBatchSize = 0;
 
-            if (image.empty()) {
-                slog::warn << "Image " + imagePaths[inputIndex] + " cannot be read!" << slog::endl;
-                continue;
-            }
+    std::string image_tensor_name;
+    size_t image_tensor_width = 0;
+    size_t image_tensor_height = 0;
+    ov::Layout image_tensor_layout;
+    ov::Shape image_tensor_shape;
+    ov::element::Type image_tensor_type;
+
+    std::string info_tensor_name;
+    ov::Layout info_tensor_layout = {"?C"};
+    ov::Shape info_tensor_shape;
+
+    for (ov::Output<ov::Node> input : inputs) {
+        ov::Shape shape = input.get_shape();
+        size_t dims = shape.size();
 
-            images.push_back(image);
+        if (dims == 2) {
+            info_tensor_name = input.get_any_name();
         }
-        if (images.empty()) throw std::logic_error("Valid input images were not found!");
+        if (dims == 4) {
+            image_tensor_name = input.get_any_name();
+            image_tensor_type = input.get_element_type();
+            image_tensor_shape = shape;
+            image_tensor_layout = ov::layout::get_layout(input);
+            if (image_tensor_layout.empty() && image_tensor_type == ov::element::f32)
+                image_tensor_layout = {"NCHW"};
+            if (image_tensor_layout.empty() && image_tensor_type == ov::element::u8)
+                image_tensor_layout = { "NHWC" };
```

Comment:
Is it possible to set `{ "NHWC" }` in `ppp` for all cases?

Reply:
should be possible, I leave this as an example that old models can work with not changed demo. But question is valid, probably no much sense to keep this and move to use capabilities provided by pre-post processor

Reply:
made nhwc, u8 as default input